src/executor_config/barrier.md

```diff
@@ -0,0 +1,75 @@
+# Barrier
+Before any XCMs are executed in the XCM executor, they must pass through the Barrier. The Barrier type implements the [`ShouldExecute`] trait and serves as the firewall for the xcm-executor. Each time the xcm-executor receives an XCM, it consults the Barrier to determine if the XCM should be executed.
```

Comment:
I'd add in a new paragraph that barriers can also modify the message, like adding a new instruction or whatnot.